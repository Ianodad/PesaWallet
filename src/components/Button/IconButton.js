import React from 'react';
import {StyleSheet, View, Image, TouchableOpacity} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome5';
// import colors from '../../config/colors';
import defaultStyles from '../../config/styles';
// import Text from '../Text';
// import Info from './Info';

const IconButton = ({
  image,
  style,
  imageStyle,
  info,
  iconStyle,
  icon,
  iconSize,
  color = 'primary',
  onPress,
}) => {
  // console.log(detail)
  return (
    <TouchableOpacity
      style={[
        styles.button,
        style,
        {backgroundColor: defaultStyles.colors[color]},
      ]}
      onPress={onPress}>
      {icon && (
        <Icon style={[styles.icon, iconStyle]} name={icon} size={iconSize} />
      )}

      {image && (
        <Image
          style={[styles.logo, imageStyle]}
          resizeMode="contain"
          source={image}
        />
      )}
      {info}
    </TouchableOpacity>
  );
};

export default IconButton;

const styles = StyleSheet.create({
  button: {
    flex: 1,
    borderRadius: 10,
    justifyContent: 'center',
    alignItems: 'center',
  },
  logo: {
    flex: 1,
  },
  subTitle: {
    position: 'absolute',
    zIndex: 2,
  },
  info: {
    flex: 1,
    position: 'absolute',
    zIndex: 2,
  },
});
