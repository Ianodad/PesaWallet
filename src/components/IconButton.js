import React from 'react';
import {StyleSheet, Text, View, Image, TouchableOpacity} from 'react-native';
import colors from '../config/colors';

const IconButton = ({image, style, logoStyle, color = 'primary', onPress}) => {
  return (
    <TouchableOpacity
      style={[styles.button, style, {backgroundColor: colors[color]}]}
      onPress={onPress}>
      <Image style={[styles.logo, logoStyle]} resizeMode="contain" source={image} />
    </TouchableOpacity>
  );
};

export default IconButton;

const styles = StyleSheet.create({
  button: {
    flex: 1,
    borderRadius: 10,
    justifyContent: 'center',
    alignItems: 'center',
  },
  logo: {
    flex: 1,
  },
});
