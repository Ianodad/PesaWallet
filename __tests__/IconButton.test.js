import React from 'react';
import IconButton from '../src/components/Button/IconButton';

const defaultProps = {
  title: 'title',
  onPress: jest.fn(),
  style: {margin: '10px'},
  imageStyle: {margin: '10px'},
  image: jest.mock('../src/assets/bootsplash_logo.png'),
  info: 'test',
  textStyle: {margin: 'black'},
  iconStyle: {margin: '10px'},
  color: 'primary',
  buttonType: true,
};

describe('<IconButton/>', () => {
  const fn = jest.fn();
  const image = jest.mock('../src/assets/bootsplash_logo.png');
  describe('Rendering with snapshot', () => {
    // should render with any outcome
    it('should match to snapshot', () => {
      const component = shallow(<IconButton />);
      expect(component).toMatchSnapshot();
    });
    //
    it('should render with title snapshot', () => {
      const component = shallow(<IconButton title={'Submit'} />);
      expect(component).toMatchSnapshot();
    });
    it('should render with image', () => {
      const component = shallow(<IconButton title={'Submit'} image={image} />);
      expect(component).toMatchSnapshot();
    });
     it('should render with icon', () => {
      const component = shallow(<IconButton title={'Submit'} icon={true} />);
      expect(component).toMatchSnapshot();
    });

    it('should render with function passed snapshot', () => {
      const component = shallow(<IconButton title={'Submit'} onPress={fn} />);
      expect(component).toMatchSnapshot();
    });

    it('should render with info passed', () => {
      const component = shallow(
        <IconButton title={'Submit'} info={'test case'} />,
      );
      expect(component).toMatchSnapshot();
    });

    it('should render with with passed props', () => {
      const component = shallow(<IconButton {...defaultProps} />);
      expect(component).toMatchSnapshot();
    });
  });
});
