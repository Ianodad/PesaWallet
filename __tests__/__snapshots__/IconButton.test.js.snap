// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton/> Rendering with snapshot should match to snapshot 1`] = `ShallowWrapper {}`;

exports[`<IconButton/> Rendering with snapshot should render with function passed snapshot 1`] = `ShallowWrapper {}`;

exports[`<IconButton/> Rendering with snapshot should render with icon 1`] = `ShallowWrapper {}`;

exports[`<IconButton/> Rendering with snapshot should render with image 1`] = `ShallowWrapper {}`;

exports[`<IconButton/> Rendering with snapshot should render with info passed 1`] = `ShallowWrapper {}`;

exports[`<IconButton/> Rendering with snapshot should render with title snapshot 1`] = `ShallowWrapper {}`;

exports[`<IconButton/> Rendering with snapshot should render with with passed props 1`] = `ShallowWrapper {}`;
